packages/graphback-datasync/src/DataSyncPlugin.ts

```diff
@@ -1,37 +1,39 @@
-import { createVersionedInputFields,createVersionedFields, createInputTypeForScalar } from "@graphback/codegen-schema";
-import { GraphQLNonNull, GraphQLSchema, buildSchema, GraphQLString, GraphQLResolveInfo, GraphQLInt } from 'graphql';
-import { GraphbackCoreMetadata, GraphbackPlugin, ModelDefinition, getInputTypeName, GraphbackOperationType, parseRelationshipAnnotation, metadataMap, GraphbackContext, getSelectedFieldsFromResolverInfo, GraphbackTimestamp } from '@graphback/core'
-import { parseMetadata } from "graphql-metadata";
-import { SchemaComposer, ObjectTypeComposerFieldConfig } from 'graphql-compose';
+import { GraphQLNonNull, GraphQLSchema, buildSchema, GraphQLResolveInfo, GraphQLInt, GraphQLBoolean, GraphQLList } from 'graphql';
+import { GraphbackCoreMetadata, GraphbackPlugin, ModelDefinition, getInputTypeName, GraphbackOperationType, parseRelationshipAnnotation, GraphbackContext, getSelectedFieldsFromResolverInfo, GraphbackTimestamp, isSpecifiedGraphbackScalarType } from '@graphback/core'
+import { SchemaComposer, ObjectTypeComposerFieldConfig, ObjectTypeComposer } from 'graphql-compose';
 import { IResolvers, IFieldResolver } from '@graphql-tools/utils'
 
 import { getDeltaType, getDeltaListType, getDeltaQuery } from "./deltaMappingHelper";
 import { isDataSyncService, isDataSyncModel, DataSyncFieldNames } from "./util";
 
-/**
- * Configuration for Schema generator CRUD plugin
- */
-
-
 export const DATASYNC_PLUGIN_NAME = "DataSyncPlugin";
 
+export interface DataSyncPluginConfig {
+  useVersion: boolean
+}
+
 /**
  * DataSync plugin
  *
  * Plugin is enabled by """ @datasync """ annotation
  * It will generate delta queries
  */
 export class DataSyncPlugin extends GraphbackPlugin {
-  protected addVersion: boolean;
+  protected useVersion: boolean;
 
-  public constructor(addVersion: boolean = false) {
+  public constructor(config: DataSyncPluginConfig = {
+    useVersion: false
+  }) {
     super()
-    this.addVersion = addVersion
+    this.useVersion = config.useVersion
```

Comment:
It will be neater to keep all config in a single global variable.

```suggestion
    this.config = config
```